### DISCUSSION TITLE
What is the proper way to send event with payload when using useActor in react typescript?

### DISCUSSION THREAD

**Discussion by edwardccg** [statelyai/xstate#2352] [Q&A] [answered]
Hi, 
First of all, thank you for this amazing library!

I started a new project and using the latest version of xstate:
- xstate@4.20.1
- @xstate/react@1.4.0

There's a deprecation message for `useService`, so i was trying to use `useActor`. However, I notice the `send()` function do not really have the same signature with the previous `send()` provided by the `useService`

This is also the first time I try to use typescript, I am having some type warning by the IDE as well. Please advise what would be the correct approach for sending event + payload to the machine for `useActor`

Please find the replication in the code sandbox below. When the payload message did send over, it will be shown in the UI.
https://codesandbox.io/s/xstate-react-typescript-useactor-euu6z?file=/src/ChildComponent.tsx

```javascript
const [current, sendActor] = useActor(actorService);
const [, sendService] = useService(actorService); //this is being deprecated

//This doesn't work, the payload message does not send it to machine
sendActor("TOGGLE", { payload: "from Actor 1" })

//This works, but IDE is throwing type warning of 'payload' does not exists in type 'EventObject'
sendActor({ type: "TOGGLE", payload: "from Actor 2" })

//Following are the original useService's send function: 
// This works, but IDE is throwing type warning of 'string' is not assignable to type 'never'
sendService("TOGGLE", { payload: "from Service 1" })

//This works, but IDE is throwing type warning of 'payload' does not exists in type 'EventObject'
sendService({ type: "TOGGLE", payload: "from Service 2" })
```

**Comment by davidkpiano** [comment, marked as answer]
The type warnings are correct ✔️, since the events aren't typed in the machine currently. You need to add those types:

```ts
export type PropertyPanelProps = {
  actorService: Interpreter<ToggleContext, any, ToggleEvents>;
};
```

See here: https://codesandbox.io/s/xstate-react-typescript-useactor-forked-cpcqk?file=/src/ChildComponent.tsx

**Reply by edwardccg** [reply]
Thanks David. The type warning is resolved!

For my other questions. When using `useService`, I can send payload like 
```javascript 
sendService("TOGGLE", { payload: "from Service 1" })
``` 
but in `useActor`, the following doesn't seems to work although the type definition for sendActor did suggest a 2nd arguments of `payload?: EventData`. What would be the usage for that?
```javascript
sendActor("TOGGLE", { payload: "from Actor 1" })
```
